cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Insert.hs

```diff
@@ -197,11 +197,11 @@ insertShelleyBlock syncEnv shouldLog withinTwoMins withinHalfHour blk details is
 
     insertStakeSlice syncEnv $ apStakeSlice applyResult
 
-    when (ioGov iopts)
+    when (ioGov iopts && (withinHalfHour || unBlockNo (Generic.blkBlockNo blk) `mod` 10000 == 0))
```

Comment:
`(withinHalfHour || unBlockNo (Generic.blkBlockNo blk) `mod` 10000 == 0)` could probably sit in the where clause